main/src/ca/uwaterloo/flix/language/dbg/AstPrinter.scala

```diff
@@ -30,91 +30,72 @@ object AstPrinter {
-      if (asts.contains("EffectBinder")) writeToDisk("EffectBinder", formatReducedAst(flix.getEffectBinderAst))
-      if (asts.contains("Eraser")) writeToDisk("Eraser", formatReducedAst(flix.getEraserAst))
-      if (asts.contains("Reducer")) writeToDisk("Reducer", formatReducedAst(flix.getReducerAst))
-      if (asts.contains("VarOffsets")) writeToDisk("VarOffsets", formatReducedAst(flix.getVarOffsetsAst))
-      if (asts.contains("JvmBackend")) () // wip
-    }
+    val optionPhases = flix.options.xprintphase
+    val shouldPrintEverything = optionPhases.contains("all") || optionPhases.contains("All")
+    val phaseMap = if (shouldPrintEverything) allPhases(includeUnfinished = false)
+                   else allPhases().filter(pair => optionPhases.contains(pair._1))
+    printPhaseMap(phaseMap)
   }
 
   /**
     * Writes all the formatted asts to disk.
     */
   def printAllAsts()(implicit flix: Flix): Unit = {
-    //
-    // Front-end
-    //
-
-    // Parser wip
-    // Weeder wip
-    // Desugar wip
-    // Namer wip
-    // Resolver wip
-    // Kinder wip
-    // Deriver wip
-    // Typer wip
-    // Entrypoint wip
-    // PredDeps wip
-    // Stratifier wip
-    // PatMatch wip
-    // Redundancy wip
-    // Safety wip
-
-    //
-    // Back-end
-    //
-
-    writeToDisk("Lowering", formatLoweredAst(flix.getLoweringAst))
-    writeToDisk("TreeShaker1", formatLoweredAst(flix.getTreeShaker1Ast))
-    writeToDisk("MonoDefs", formatLoweredAst(flix.getMonoDefsAst))
-    writeToDisk("MonoTypes", formatLoweredAst(flix.getMonoTypesAst))
-    writeToDisk("Simplifier", formatSimplifiedAst(flix.getSimplifierAst))
-    writeToDisk("ClosureConv", formatSimplifiedAst(flix.getClosureConvAst))
-    writeToDisk("LambdaLift", formatLiftedAst(flix.getLambdaLiftAst))
-    writeToDisk("Tailrec", formatLiftedAst(flix.getTailrecAst))
-    writeToDisk("Optimizer", formatLiftedAst(flix.getOptimizerAst))
-    writeToDisk("TreeShaker2", formatLiftedAst(flix.getTreeShaker2Ast))
-    writeToDisk("EffectBinder", formatReducedAst(flix.getEffectBinderAst))
-    writeToDisk("Eraser", formatReducedAst(flix.getEraserAst))
-    writeToDisk("Reducer", formatReducedAst(flix.getReducerAst))
-    writeToDisk("VarOffsets", formatReducedAst(flix.getVarOffsetsAst))
+    printPhaseMap(allPhases(includeUnfinished = false))
+  }
+
+  /**
+    * Goes through each map binding and calls [[writeToDisk]].
+    */
+  private def printPhaseMap(phaseMap: Map[String, () => String])(implicit flix: Flix): Unit = {
+    for ((phase, printer) <- phaseMap)
+      writeToDisk(phase, printer())
+  }
+
+  /**
+    * Returns a list of map the phases of flix along with a thunked pretty printed AST.
+    *
+    * Returns only the phases that can be pretty printed if `includeUnfinished` is false.
+    */
+  def allPhases(includeUnfinished: Boolean = true)(implicit flix: Flix): Map[String, () => String] = {
```

Comment:
This bool flag  is not the prettiest, but when crashing or printing all asts, we don't want a bunch of "wip" files, but we do want them here when ShowAstProvider is asking us to print "Parser" (so we don't give "Lowering" or whatever the closest string match is)